app/services/update_user_service.rb

```diff
@@ -0,0 +1,23 @@
+class UpdateUserService
+  def self.call(user, method)
+    new(user, method).update_user
+  end
+
+  def initialize(user, method)
+    @user = user
+    @method = method
+  end
+
+  def update_user
+    @user.__send__(@method)
+  rescue JsonApiClient::Errors::ClientError, JsonApiClient::Errors::ServerError => e
+    e.env.body.delete("traces")
+    Raven.extra_context(e.env)
+
+    if e.is_a? JsonApiClient::Errors::ClientError
+      Raven.capture_exception(e)
+    else
+      raise e
+    end
```

Comment:
Good call moving this out of the controller, although it feels like we should move this somewhere so we can avoid even needing a specialized service ... maybe hook this into `JsonApiClient` somehow. Something to keep in mind for later.

Reply:
Yeah good idea